### ISSUE TITLE
Sensor data doesn't transfer to NS site

### ISSUE THREAD

**Issue by olaluukkainen** [pazaan/600SeriesAndroidUploader#284]
* Uploader Version Number: 0.7.1 
* Your Android Phone Model Name (e.g. Samsung J1 Ace): Samsung S5 and S7
* Android Version Number (e.g. use 4.4 for 4.4.4): 8.0.0 (S7) 6.0.1 (S5)
* Network Connection at the time the issue occurred (Wi-Fi, Mobile): Both mobile and Wi-Fi
* (if relevant) Nightscout website version (e.g. 0.8.4 Funnel cake, master / dev branch): 14.0.6

All of a sudden both of our uploader phones stopped sending sensor data. Treatments, micro boluses, IOB, pump info etc. is visible in NS site. No errors in uploader phones. Sensor data is shown correctly in uploader phones.

No error codes.

#### Severity Score
4

**Comment by Pogman** [comment]
Could be an issue with your NS database being near full. Go into your NS admin page and clear out some of the old data.

**Comment by olaluukkainen** [comment]
Don't know what the problem was but both uploaders started to work after re-installing the software. Sensor data still doesn't show in NS from the blackout period.

**Comment by olaluukkainen** [comment]
So this is still an issue. The problem started again today on both phones.

**Comment by pazaan** [comment]
@olaluukkainen Is this still occurring? If so, I'd need to set up some time so that we can remote debug this together. I no longer have 600 Series sensors, so I'm unable to test myself.

**Comment by funkstille** [comment]
Today both of my phones had the same issue. No data is transferred to nightscout anymore. 

Switched the phone, same problem 
Updated nightscout on heroku to latest, same problem 
Downgraded uploader from 0.73 to 0.71, same problem 

I switched from mongo to atlas round about 2 months ago, so that should not be the root cause as well. Database size is 108 MB ~ 21% of the free Space. Any suggestions?

**Comment by Pogman** [comment]
Make sure your using the latest version of Nightscout currently 14.0.7

**Comment by funkstille** [comment]
Yes, upgraded to nightscout 14.0.7 after the problem occurred. Did not solve the problem. Workaround is forwarding to xDrip and uploading with xDrip

**Comment by Pogman** [comment]
There were issues pre-14.0.7 that would not handle cgm backfills and other cgm issues. Have you tried again just running with an uploader reinstall done after 14.0.7 ?

**Comment by funkstille** [comment]
I did a upgrade to 14.0.7 and since that had no effect I have done a reinstall with downgrade to 0.71. Did not work. But since it is pretty late in Germany I might need a few hours of sleep. I will have look tomorrow morning, the backup phone has still 0.73 installed. 

But everything worked fine until tonight. Even with nightscout lower then 14.0.7 and uploader on 0.73.

**Comment by funkstille** [comment]
I purged the uploader on both phones and reinstalled version 0.73.
Still no cgm data upload. Other data like ‘usb device failure’ is uploaded.
I also recognised that the app crashes from time to time.
At the moment I am using drip+ as uploader.

Should I provide any logs or other data? Screensharing?

**Comment by Pogman** [comment]
0.7.2 and 0.7.3 are have crashes for some people and that should get sorted soon.

I don't know what could be wrong with your cgm data specially if your seeing the issue with 0.7.1 too.

Can you go into the debug settings and enable "extended errors" Go back to the log screen and you should see items like this:

![Screenshot 2020-12-05 131023](https://user-images.githubusercontent.com/7413594/101243939-5a068500-36fb-11eb-9953-d62b7287c7b0.jpg)

The nightscout info tells us what happened after a successful upload. The E:1/0/1 part is for cgm entries and there should be something similar for you after each 5 min poll.

Also look back through the log for this line:

![Screenshot 2020-12-05 131632](https://user-images.githubusercontent.com/7413594/101244068-2415d080-36fc-11eb-98a7-3db956fc5a6f.jpg)

This will show the exact time difference and version that the uploader is seeing. Can you check yours?

**Comment by funkstille** [comment]
Thanks you for your reply. This morning at about 11 o’clock it started to work again. Until that time I haven’t had the logs switched to extended errors. So sadly there is nothing to share from before. Right now the time difference between my nightscout instance 14.0.7 changes every now and then but is always less then 60 seconds. If the problem appears again I will log the information!

**Comment by Pogman** [comment]
The extended errors option just toggles what is visible and will always show the last few days whenever you turn is on.

The NS time difference will always vary but it should be a small amount and well under 5 minutes. So that looks good and not causing your issue. 

Do you run multiple uploaders? This should not be a problem and even running simultaneously but maybe there's a clue.

**Comment by funkstille** [comment]
![462E3C77-5140-453D-BF20-70B2FFF17CE9](https://user-images.githubusercontent.com/13382630/101407856-ccce5680-38db-11eb-83aa-0e440311c1be.jpeg)
![D54FC39B-85A9-4C9C-8D8E-B89A80349CCC](https://user-images.githubusercontent.com/13382630/101407861-cdff8380-38db-11eb-8185-66c93537f8a4.jpeg)
![68EA1D18-C9D9-45E8-8C13-FF16DB4A211A](https://user-images.githubusercontent.com/13382630/101407864-cdff8380-38db-11eb-8301-a3bcbd16af31.jpeg)


Took a while. Here are the screenshots

**Comment by Pogman** [comment]
Ok it looks like it's uploading a lot of sensor readings successfully. This also indicates that you must have either installed the app very recently and/or it is doing a very large backfill.

Are you still seeing no sensor readings in nightscout?

**Comment by funkstille** [comment]
No. The problem occurred only from 19.45 until aprox. 11 o’clock. After that everything worked fine again, with some app crashes now and then

**Comment by Pogman** [comment]
I don't really know what has happened for you here with your missing period. When you reinstalled or updated the app it should have checked and resent and missing data for you. When the app crashes it should still keep collecting and sending data though you would need to click the app icon to see the UI.

Other things that come to mind are that the pump had no valid sensor readings during that period or maybe even some glitch with nightscout or it's database provider.

Sure keep an eye on it and let us know if it starts having issues again. It might be easier to catch the cause when the issue is live.

**Comment by funkstille** [comment]
That’s absolutely fine with me. Thanks for your support. I will get in touch if the problem occurs again

**Comment by rizso** [comment]
I have now the same problem. We use 2 uploader phone and suddenly one of these phones doesn't transfer the sensor data to NS site. IOB, COB, basal profile, pump information,... are visible on NS site, only the sensor data doesn't.
What could be the problem?

**Comment by Pogman** [comment]
Try uninstalling and reinstalling the app and see how it goes.

**Comment by rizso** [comment]
I did it, but no change. In fact, since last night, the other uploader phone isn't tranfering the sensor data. The other information is still visible on NS site.
